[8.16] Updates docs for and related to the `excludedDataTiersForRuleExecution` advanced setting (#5962)
elastic/security-docs: docs/detections/detection-engine-intro.asciidoc

In addition, the following support restrictions are in place:

> * Indicator match rules don't support cold or frozen data but will query cold and frozen {ref}/data-tiers.html[data tiers] if they match the rule's index pattern. To exclude query results from cold and frozen tiers, create a <<exclude-cold-frozen-data-individual-rules,Query DSL filter>> and add it to individual rules. 

Reviewer: This part still concerns me because it feels like we're telling users that cold and frozen data aren't supported but the rules by default will put them in this unsupported situation. I think the key points I'd like to communicate are:
- Indicator match rules don't support cold and frozen data
- Normally (when shards are available and timestamps in the data are correct), cold and frozen data should be excluded efficiently by timestamp (i.e. as long as the cold and frozen data is older than the rule interval + additional lookback, it should be fine)
- In abnormal circumstances (shards unavailable and/or bad timestamps) the _tier filter can be added to explicitly exclude cold/frozen data and resolve issues, although for 8.16.0 the _tier filter still isn't sufficient to fix IM rules with unavailable frozen shards so I wouldn't put too much detail about this

Maybe we can write something like "Indicator match rules don't support cold or frozen data. Since rules query the last X=(rule interval + lookback) minutes, indices queried by the rule should wait at least X minutes before moving to cold or frozen tier. Alternatively, to explicitly exclude query results from cold and frozen...".

Reviewer 2: Chatting with Marshall. What do you think if we go with something like:

```suggestion
* Indicator match rules don't support cold or frozen data. Cold or frozen data in indices queried by Indicator Match rules must be older than the time range queried by the rule. If your data's timestamps are unreliable, you can exclude cold and frozen tier data using a <<exclude-cold-frozen-data-individual-rules,Query DSL filter>>. 
```

Please feel free to modify the language to be more grammatically correct :)

In addition, the following support restrictions are in place:

> * Indicator match rules don't support cold or frozen data, but will query cold and frozen {ref}/data-tiers.html[data tiers] if they exist. To exclude query results from cold and frozen tiers, configure the `excludedDataTiersForRuleExecution` <<exclude-cold-frozen-data-rule-executions,advanced setting>> (which applies to all rules in a space), or add a <<exclude-cold-frozen-data-individual-rules,Query DSL filter>> to individual rules. 

Reviewer: `if they exist` -> `if they match the rule's index pattern` ?

Reviewer 3: I think this should be updated to remove the reference to the "advanced setting" as it's not helpful in the context of IM rules. For IM rules, what helps the user is the workflow for excluding tiers for **SINGLE** rules, not for **ALL** rules.

Author: ```suggestion
* Indicator match rules don't support cold or frozen data but will query cold and frozen {ref}/data-tiers.html[data tiers] if they match the rule's index pattern. To exclude query results from cold and frozen tiers, create a <<exclude-cold-frozen-data-individual-rules,Query DSL filter>> and add it to individual rules. 
```

+

> IMPORTANT: When the `excludedDataTiersForRuleExecution` advanced setting is enabled, some rule types (specifically the indicator match, event correlation, and {esql} rule types) might fail if a frozen or cold shard that matches the rule's specified index pattern is unavailable during rule executions. The best path forward continues to be modifying the index patterns to only use hot tier data.

Reviewer 2: ```suggestion
IMPORTANT: When the `excludedDataTiersForRuleExecution` advanced setting is enabled, some rule types (specifically the indicator match, event correlation, and {esql} rule types) might fail if a frozen or cold shard that matches the rule's specified index pattern is unavailable during rule executions. The best path forward continues to be modifying the index patterns to only match data in the tiers you would like to target.
```

Reviewer: ```suggestion
IMPORTANT: When the `excludedDataTiersForRuleExecution` advanced setting is enabled, indicator match, event correlation, and {esql} rules might fail if a frozen or cold shard that matches the rule's specified index pattern is unavailable during rule executions. The best path forward continues to be modifying the index patterns to only use hot tier data.
```
++ to Yara's suggestion also

Author: Made a slight edit to the last sentence ^. Let me know if it's still correct or needs to be reverted.

cc: @yctercero @marshallmain
